buildSrc/src/main/groovy/io/micronaut/guides/OpenAPIGenerator.groovy

```diff
@@ -67,4 +60,56 @@ class OpenAPIGenerator {
             throw new GradleException("OpenAPI generator failed with \"" + e.message + "\"");
         }
     }
+
+    private static CodegenConfigurator createConfiguration(File definitionFilePath,
+                                                           GuideMetadata.OpenAPIGeneratorConfig config,
+                                                           File destination,
+                                                           TestFramework testFramework,
+                                                           BuildTool buildTool,
+                                                           String destinationPackage) {
+        CodegenConfigurator configurator = new CodegenConfigurator();
+        configurator.setInputSpec(definitionFilePath.getPath());
+        configurator.setGeneratorName(config.generatorName)
+        configurator.setOutputDir(destination.getPath())
+        configurationAdditionalProperties(config, testFramework, buildTool, destinationPackage).each { k, v ->
+            configurator.addAdditionalProperty(k, v);
+        }
+        configurator
+    }
+
+    @NonNull
+    private static Map<String, Object> configurationAdditionalProperties(@NonNull GuideMetadata.OpenAPIGeneratorConfig config,
+                                                                         @NonNull TestFramework testFramework,
+                                                                         @NonNull BuildTool buildTool,
+                                                                         @NonNull String destinationPackage) {
+        Map<String, Object> additionalProperties = [:]
+        testProperty(testFramework).ifPresent(value -> additionalProperties.put(TEST, value))
+        buildProperty(buildTool).ifPresent(value -> additionalProperties.put(PROPERTY_BUILD, value))
+        additionalProperties.put(PROPERTY_CONTROLLER_PACKAGE, destinationPackage + PACKAGE_CONTROLLER)
+        additionalProperties.put(PROPERTY_API_PACKAGE, destinationPackage + PACKAGE_API)
+        additionalProperties.put(PROPERTY_MODEL_PACKAGE, destinationPackage + PACKAGE_MODEL)
+        additionalProperties.put(PROPERTY_GENERATOR_NAME, GuideMetadata.OpenAPIGeneratorConfig.GENERATOR_JAVA_MICRONAUT_SERVER)
```

Comment:
Should probably be `config.generatorName` instead of `GuideMetadata.OpenAPIGeneratorConfig.GENERATOR_JAVA_MICRONAUT_SERVER`.
There is also `java-micronaut-client` that can be used in the guides, so the property is needed.

Reply:
> Should probably be config.generatorName instead of GuideMetadata.OpenAPIGeneratorConfig.GENERATOR_JAVA_MICRONAUT_SERVER

You are right. Please, do the change.